src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,201 @@
+            executeAddTodo(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_ADD_DEADLINE:
+            executeAddDeadline(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_ADD_EVENT:
+            executeAddEvent(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_LIST:
+            executeList();
+            break;
+        case COMMAND_MARK_DONE:
+            executeMarkDone(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_DELETE:
+            executeDelete(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_EXIT:
+            executeExit();
+            break;
+        default:
+            throw new DukeException();
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    /**
+     * Executes exiting program.
+     */
+    private static void executeExit() throws IOException {
+        ui.printGoodBye();
+        System.exit(0);
+    }
+
+    /**
+     * Executes marking task as done.
+     *
+     * @param commandArgs Processed input arguments from user.
+     */
+    private static void executeMarkDone(String commandArgs) {
+        int taskIndex = Integer.parseInt(commandArgs.strip()) - 1;
+        Task currentTask = tasks.getTaskByIndex(taskIndex);
+        currentTask.markAsDone();
+        ui.printConfirmMarkDone(currentTask);
+    }
+
+    private static void executeDelete(String commandArgs) {
+        int taskIndex = Integer.parseInt(commandArgs.strip()) - 1;
+        ui.printConfirmDelete(tasks.getTaskByIndex(taskIndex), tasks.getNumberOfTasks());
+        tasks.removeTaskByIndex(taskIndex);
+    }
+
+    /**
+     * Executes listing tasks.
+     */
+    private static void executeList() {
+        System.out.println(HORIZONTAL_LINE);
+        if (tasks.getNumberOfTasks() > 0) {
+            System.out.println("\t" + "Here are the tasks in your list:");
+            for (int i = 0; i < tasks.getNumberOfTasks(); i++) {
+                System.out.println("\t" + (i + 1) + ". " + tasks.getTaskByIndex(i));
+            }
+        } else {
+            System.out.println(MESSAGE_NO_TASK);
+        }
+    }
+
+    /**
+     * Executes adding an event.
+     *
+     * @param commandArgs Processed input arguments from user.
+     */
+    private static void executeAddEvent(String commandArgs) throws IllegalEventException {
+        int indexOfTimePrefix = commandArgs.indexOf(PREFIX_TIME_EVENT);
+        if (indexOfTimePrefix != -1) {
+            String description = commandArgs.substring(0, indexOfTimePrefix).trim();
+            String time = commandArgs.substring(indexOfTimePrefix + 3).trim();
+            tasks.addTask(new Event(description, time));
+            ui.printConfirmAdd(tasks.getTaskByIndex(tasks.getNumberOfTasks() - 1), tasks.getNumberOfTasks());
```

Comment:
This line is a bit long, and you also repeated the function call for `tasks.getNumberOfTasks()`. You can extract this into a variable eg. `numberOfTasks` and reuse it.